feature/json-formatted-logs

dtitov opened the pull request:
![](https://habrastorage.org/webt/cy/sw/nu/cyswnues_cblirkwewacsgysw0i.png)

silverdaz commented on .travis.yml:
```diff
@@ -10,7 +10,7 @@ before_install:
         make pull
         make common
-        make -j 4 images
+        make -j 1 images
```
Remove the flag instead of making it 1.

But anyway, take away that commit from that PR, it belongs to another one.

silverdaz commented on lega/utils/logging.py:
```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+
+from logging.handlers import SocketHandler
+import ssl
```
Not used.
Codacy complains.

silverdaz commented on lega/utils/logging.py:
```diff
@@ -0,0 +1,12 @@
+import ssl
+
+
+class LogstashHandler(SocketHandler):
```
I don't believe making a module and a class, to overload _one_ single function is the right solution.

For testing around, do what you feel like, but I don't think it should be commited that way.

silverdaz commented on lega/conf/__init__.py:
```diff
@@ -2,6 +2,7 @@
 import configparser
 import logging
 from logging.config import fileConfig, dictConfig
+import lega.utils.logging
```
from lega.utils import logging

dtitov commented:
I fixed Makefile because it was working wrong all that time with Pull Requests (as a lucky coincidence it didn't result in a red build). Answers to the rest of comments are below.


> from lega.utils import logging

No, it fails like that. Should be `import lega.utils.logging`

> Not used.

Fixed.

> I don't believe making a module and a class, to overload one single function is the right solution.
> For testing around, do what you feel like, but I don't think it should be commited that way.

Honestly, I believe our code is very unstructured. I didn't complain before because I didn't work much with the LocalEGA Python code, but now I have to deal with it, so I suggest to spend some time (maybe next sprint) to perform a refactoring in an OOP manner. And this `LogstashHandler` is the first step towards it.

silverdaz commented:
I approve the PR.

We can talk about the comments in the standup, or the sprint review.
